ui/models.py

```diff
@@ -16,12 +17,29 @@
 from ui import utils
 from ui.constants import VideoStatus, YouTubeStatus, StreamSource
 from ui.encodings import EncodingNames
-from ui.tasks import delete_s3_objects
-
+from ui.utils import get_bucket
 
 TRANSCODE_PREFIX = 'transcoded'
 
 
+@shared_task(bind=True)
+def delete_s3_objects(self, bucket_name, key, as_filter=False):  # pylint:disable=unused-argument
+    """
+    Delete objects from an S3 bucket
+
+    Args:
+        bucket_name(str): Name of S3 bucket
+        key(str): S3 key or key prefix
+        as_filter(bool): Filter the bucket by the key
+    """
+    bucket = get_bucket(bucket_name)
+    if not as_filter:
+        bucket.delete_objects(Delete={'Objects': [{'Key': key}]})
+    else:
+        for obj in bucket.objects.filter(Prefix=key):
+            obj.delete()
+
+
```

Comment:
Seems like `tasks.py` is a better place for this than `models.py`?

Reply:
It's a task, but it's only used by these models, and it's specifically designed _for_ models. Having this in `tasks.py` also created a circular import (tasks -> api -> models -> tasks). I felt it was cleaner/more intuitive to move this into `models.py` than to inline import both the api and models in the `post_hls_to_edx` task.

Reply:
👍